### PR TITLE
Switch from Tailwind 0.7.2 to 1.0.4

### PR DESCRIPTION
**Pull request by cameron-elliott**
This is an effort to get the docs template off of Tailwind 0.7.2 and upto 1.0.4

Much of the work came from the Tailwind upgrade of the blog template by @damiani and @andypa
The rest of the work came from: 
[Tailwind 1.x Upgrade docs](https://tailwindcss.com/docs/upgrading-to-v1/)

Questions I still have:
I removed files in the repo for this path:
/source/assets/build/
Is the the best move? (doesn't seem like these build artifacts should be in repo, as they are not in blog template)

Added composer.lock per composer and Stackover Flow
Saved my package-lock.json changes, is that the best source of truth?

I have reviewed the pages visually and with some resizing for responsiveness-testing, and the pagew look good to me, even with the minor color changes between 0.7.2 and 1.0.4

Please let me know if problems are discovered or I should make some changes.

Thanks for the support so far.

### CODE REVIEW

**Comment by mattstauffer** [comment]
@cameron-elliott you keep tagging Andy.. I think you're intending to tag @anthonyterrell :)

Looking forward to reading this over later! Thanks for all of your hard work on this!

File: composer.lock [hunk 1]
```diff
@@ -0,0 +1,1827 @@
+{
```
Comment (mattstauffer) [inline composer.lock]: We'll want to keep this out of here. We want to commit `composer.lock` on projects, but not on templates/generators.

File: source/_assets/sass/_base.scss [hunk 2]
```diff
@@ -128,3 +132,8 @@ pre {
         @apply .p-0;
     }
 }
+
+::selection {
+    @apply .bg-blue-500;
+    @apply .text-white;
+}
```
Comment (mattstauffer) [inline source/_assets/sass/_base.scss]: Could you add a line ending here?

And I think we intentionally didn't style the selection, so let's drop this unless this was a part of the upgrade process?

**Comment by cameron-elliott** [comment]
Actually the original submitter of the PR to upgrade the 'blog' repo was here:  [Andypa upgrade blot repo to Tailwind 1.x](https://github.com/tightenco/jigsaw-blog-template/commit/cc9c7c82a028bc298badffc5ed740efddc70c580)

But I guess we can't at-sign mention him here, because he is not a part of this repo. 

Maybe another small selling point for merging 'blog' and 'docs' :smile:

**Comment by mattstauffer** [comment]
AHHHH YOU right!

File: composer.lock (hunk 1 above)
Comment (cameron-elliott) [reply composer.lock]: Cool, thanks for explaining.

File: source/_assets/sass/_base.scss (hunk 2 above)
Comment (cameron-elliott) [reply source/_assets/sass/_base.scss]: I will fix the line ending, thanks.

I am pretty green at some of this. I had to look up ::selection.

Anyway, [those lines came from the 'blog' repo, here](https://github.com/tightenco/jigsaw-blog-template/blob/master/source/_assets/sass/_base.scss#L136)

Diomed originally put that in, and Damiani had touched/updated it also.

Should I take it out, or leave it?

Comment (mattstauffer) [reply source/_assets/sass/_base.scss]: Ah! keep it! :) Thanks!!

**Comment by anthonyterrell** [comment]
I just pulled this down @cameron-elliott and this looks fantastic. I appreciate you taking the time to upgrade this 🎉 Thank you!

File: composer.lock (hunk 1 above)
Comment (cameron-elliott) [reply composer.lock]: Okay, composer.lock is gone

**Comment by cameron-elliott** [comment]
> I just pulled this down @cameron-elliott and this looks fantastic. I appreciate you taking the time to upgrade this Thank you!

@anthonyterrell thanks!,  @mattstauffer has been fantastic helping me :tada: , and I really appreciate you guys open sourcing such quality pieces of work.

**Comment by cameron-elliott** [comment]
> Tiny notes, and I have not built it locally yet. Will shortly. But if you could address these I'll try to build them as soon as I get a chance and then hopefully we can get it merged and ready to try out live!

@mattstauffer, Matt, when you left this message, you said "Tiny notes", does that mean you thought the commit messages were too short?

**Comment by mattstauffer** [comment]
@cameron-elliott haha, no, sorry, i meant, i had left a few tiny notes using the "suggest changes" tool. Great work on this--thanks so much!!

**Pull request merged**